project-management/src/main/java/life/qbic/projectmanagement/application/measurement/validation/MeasurementProteomicsValidator.java

```diff
@@ -276,6 +245,27 @@ ValidationResult validationProjectRelation(String sampleId, ProjectId projectId)
       return validationProjectRelation(SampleCode.create(sampleId), projectId);
     }
 
+    @PreAuthorize("hasPermission(#projectId,'life.qbic.projectmanagement.domain.model.project.Project','READ')")
+    ValidationResult validationExperimentRelation(SampleCode sampleCode, String experimentId,
+        ProjectId projectId) {
+      if (sampleInformationService.retrieveSamplesForExperiment(projectId, experimentId).stream()
+          .anyMatch(sample -> sample.sampleCode().equals(sampleCode))) {
+        return ValidationResult.successful();
+      }
+      return ValidationResult.withFailures(
+          List.of("Sample ID does not belong to this experiment: %s".formatted(sampleCode.code())));
```

Comment:
Shorter: "Unknown Sample ID: Cannot match sample to experiment"

Reply:
https://github.com/qbicsoftware/data-manager-app/pull/1302#discussion_r2362829090

Reply:
I would argue users also need to know which sample id they accidentally entered.